qemu/deps/playbook.yml

```diff
@@ -44,31 +44,28 @@
          path: '{{kar_local}}'
       register: kar_folder
 
-    - name: Clone kar
-      git:
-        repo: '{{kar_repo}}'
-        dest: '{{kar_local}}'
-      when: not kar_folder.stat.exists
-
-    - name: Run bootstrap script
-      command: ./Bootstrap.sh {{bs_options}}
-      args:
-        chdir: '{{kar_local}}'
-      when: not kar_folder.stat.exists
-      register: bootstrap_result
-      ignore_errors: True
-
-    - name: Remove kar folder when bootstrap failed
-      file:
-        path: '{{kar_local}}'
-        state: absent
-      when: bootstrap_result is failed
-
-    - name: Clean env when bootstrap failed
-      pip:
-        name: aexpect, avocado-framework, avocado-framework-plugin-vt, avocado-framework-plugin-result-html
-        state: absent
-      when: bootstrap_result is failed
+    - block:
+        - name: Clone kar
+          git:
+            repo: '{{kar_repo}}'
+            dest: '{{kar_local}}'
+          when: not kar_folder.stat.exists
+
+        - name: Run Bootstrap script
+          command: ./Bootstrap.sh {{bs_options}}
+          args:
+            chdir: '{{kar_local}}'
+          when: not kar_folder.stat.exists
+          register: bootstrap_result
+      rescue:
+         - name: Remove kar folder when bootstrap failed
+           file:
+             path: '{{kar_local}}'
+             state: absent
+         - name: Clean env when bootstrap failed 
+           pip:
+             name: aexpect, avocado-framework, avocado-framework-plugin-vt, avocado-framework-plugin-result-html
+             state: absent
 
     - fail:
```

Comment:
I think need to move this task under `rescue`

Reply:
Fixed.